src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+        }
+        File targetResource = new File(targetDir.getCanonicalPath() + relPath);
+
+        try {
+            FileUtils.copyFile(fileChanged, targetResource);
+            info("Copied file: " + fileChanged.getCanonicalPath() + " to: " + targetResource.getCanonicalPath());
+        } catch (FileNotFoundException ex) {
+            debug("Failed to copy file: " + fileChanged.getCanonicalPath());
+        } catch (Exception ex) {
+            debug(ex);
+        }
+    }
+
+    /**
+     * Deletes the corresponding file in the targetDir.
+     * 
+     * @param deletedFile the file that was deleted
+     * @param dir the directory of the deletedFile
+     * @param targetDir the corresponding targetDir of the deletedFile
+     * @param targetFileName if not null deletes the targetFile with this name
+     * @throws IOException unable to resolve canonical path
+     */
+    protected void deleteFile(File deletedFile, File dir, File targetDir, String targetFileName) throws IOException {
+        debug("File that was deleted: " + deletedFile.getCanonicalPath());
+        String relPath = deletedFile.getCanonicalPath().substring(
+                deletedFile.getCanonicalPath().indexOf(dir.getCanonicalPath()) + dir.getCanonicalPath().length());
+        if (targetFileName != null) {
+            relPath = relPath.substring(0, relPath.indexOf(deletedFile.getName())) + targetFileName;
+        }
+        File targetFile = new File(targetDir.getCanonicalPath() + relPath);
+        debug("Target file exists: " + targetFile.exists());
+        if (targetFile.exists()) {
+            targetFile.delete();
+            info("Deleted file: " + targetFile.getCanonicalPath());
+        }
+    }
+
+    /**
+     * Delete all the Java class files within the specified directory.
+     * If the directory is empty, deletes the directory as well.
+     *  
+     * @param outputDirectory the directory for compiled classes
+     */
+    protected void cleanTargetDir(File outputDirectory){
+        File[] fList = outputDirectory.listFiles();
+        if (fList != null) {
+            for (File file : fList) {
+                if (file.isFile() && file.getName().toLowerCase().endsWith(".class")) {
+                   file.delete();
+                   info("Deleted Java class file: " + file);
+                } else if (file.isDirectory()) {
+                    cleanTargetDir(file);
+                }
+            }
+        }
+        if (outputDirectory.listFiles().length == 0){
+            outputDirectory.delete();
+        }
+    }
+
+    /**
+     * Register the parent directory and all sub-directories with the WatchService
+     * 
+     * @param start parent directory
+     * @param dir path of parent directory
+     * @param watcher WatchService
+     * @throws IOException unable to walk through file tree 
+     */
+    protected void registerAll(final Path start, final Path dir, final WatchService watcher) throws IOException {
```

Comment:
Is this `dir` variable used? Seems like it is overwritten on line 693.

Reply:
It is not being overwritten. The `dir` value passed to the registerAll is the same value printed in `debug("Watching directory: " + dir.toString());`

Reply:
fixed with https://github.com/WASdev/ci.common/pull/48/commits/83e2391e82f984eb04367b51310869a7c9853607